In [21]:
print r'hello\n'

hello\n


In [1]:
import re
re.search(r'\n','\n')   # pattern then string
                       # appying a raw string to pattern does not effect, it will remain the newline

In [31]:
re.match('(.*?)=(\S+)','nfe=gdf=hrh=rj').group(2)   # searches in the beginning of the string

'gdf=hrh=rj'

In [107]:
bool(re.match('.+and','njbjand'))

True

In [116]:
re.search('n','cnjwfnejfhn').start()   # strat index
re.search('n','cnjwfnejfhn').end()     # end index, meaning next index of string end

2

In [124]:
bool(re.search('n|j','cnpjwfnejfhn'))

True

In [46]:
re.findall('print \S+','print ekvef   print n')

['print ekvef', 'print n']

# Character set

In [12]:
re.search('(\S+)\d$','cn_jwfnejf3hn2').group(1)    # \w -> alphanumeric words a-zA-Z0-9_

'cn_jwfnejf3hn'

In [22]:
re.search('\d+$','Ornado2').group()

re.sub('\d+$','','Ornado2')

'Ornado'

# Quantifiers

In [155]:
# '+' -> 1 or more
# '*' -> 0 or more
# '?' -> 0 or 1
# {n,m} -> n to m repetations=={least number of repetations, most number of repetations}

In [163]:
re.search('\w+\W+\w+','wj ojc').group()

'wj ojc'

In [208]:
stri = '10.254.254.58 - - [06/Aug/2007:00:10:05 -0700] "GET /edu/languages/google-python-class/images/puzzle/a-baaa.jpg HTTP/1.0" 200 2309 "-" "googlebot-mscrawl-moma (enterprise; bar-XYZ; foo123@google.com,foo123@google.com,foo123@google.com,foo123@google.com)"\n'
re.search('/\S*',re.search('GET\s\S*',stri).group()).group()

'/edu/languages/google-python-class/images/puzzle/a-baaa.jpg'

In [211]:
re.search('/\S*',re.search('GET\s\S*',re.search('\w*puzzle\w*',stri)).group()).group()

TypeError: expected string or buffer

In [235]:
strii = ['img.jpg','eecf']
str(''.join(strii))

'img.jpgeecf'

In [246]:
stri = '10.254.254.58 - - [06/Aug/2007:00:10:05 -0700] "GET /edu/languages/google-python-class/images/puzzle/a-baaa.jpg HTTP/1.0" 200 2309 "-" "googlebot-mscrawl-moma (enterprise; bar-XYZ; foo123@google.com,foo123@google.com,foo123@google.com,foo123@google.com)"\n'
stri.index('.jpg')
stri[52:111]

'/edu/languages/google-python-class/images/puzzle/a-baaa.jpg'

In [274]:
d = {'p-baah-bbic.jpg':'-bbic.jpg','p-babd-bbfj.jpg':'-bbfj.jpg','p-babf-bajg.jpg':'-bajg.jpg'}
for i in d:
    print re.search('-\w\w\w\w.jpg',i).group()
sorted(d.items(), key=lambda x : x[1])

-bbfj.jpg
-bbic.jpg
-bajg.jpg


[('p-babf-bajg.jpg', '-bajg.jpg'),
 ('p-babd-bbfj.jpg', '-bbfj.jpg'),
 ('p-baah-bbic.jpg', '-bbic.jpg')]

In [275]:
final_urls = ['p-baah-bbic.jpg','p-babd-bbfj.jpg','p-babf-bajg.jpg']

In [292]:
d = {}
for i in final_urls:
    d[i] = re.search('\w\w\w\w.jpg',i).group()

l = sorted(d.items(), key=lambda x : x[1])
for i in l:
    print i[0]

p-babf-bajg.jpg
p-babd-bbfj.jpg
p-baah-bbic.jpg


In [34]:
import re
s = 'delete   FROM table ejnnfc'
ss = " ".join(s.split()).lower()
print ss
st = re.search('from (\S+)',re.search('delete from \S+'," ".join(s.split()).lower()).group()).group(1)
st

delete from table ejnnfc


'table'

In [155]:
import re
s = "sys.path.insert(0,'insert')"
t = re.search('sys.path.*',s)
f = " ".join(t.group().split())
print f
g = re.search('\((.*)\)',f).group(1)
g

sys.path.insert(0,'vewvewvwe')


"0,'vewvewvwe'"

In [250]:
ss = '''#!fcjij
'''
re.search('\n+(.*)',ss).group(1)

''

In [247]:
import re
import sys
import fileinput

import_list = []

for line in fileinput.FileInput("/home/sa.goyal/Desktop/file.txt", inplace=1):
    if line is None:
        break
        
    importSearch = re.search('import.*',line)
    sysSearch = re.search('sys.path.*',line)
    nextline = re.search('\n(.*)',line)
    if nextline:
        print nextVar.group(1) + 'fc3f'
        #sys.stdout.write(line)
    '''if nextVar == 'imp' and nextVar == 'sys':
        if importSearch:
            import_list.append(importSearch.group())
        if sysSearch:
            import_list.append(sysSearch.group())
    else:
        if "sys.path.insert(0,'insert')" not in import_list:
            line = "sys.path.insert(0,'insert')\nimport logger\n"
        else:
            if "import logger" not in import_list:
                line = "import logger"'''
         
    #t = re.search('print.*',line)
    #if t:
    #    tb = " ".join(t.group().split())
    #    line = line.replace(t.group(),"logger.info({ss})".format(ss=re.search(' (.*)',tb).group(1)))
    #sys.stdout.write(line)

In [3]:
import re
import sys
import fileinput

imp = []
flag = False
with open("/home/sa.goyal/Desktop/file.txt", 'r') as fh:
    lines = fh.readlines()
    
for f in lines:
    s = re.search('import.*|sys.path.*',f)
    if s:
        imp.append(s.group())

if "sys.path.insert(0,'insert')" not in imp and "sys.path.append('insert')" not in imp:
    l1 = "sys.path.insert(0,'insert')\nimport logger\n"
    flag = True
elif "import logger" not in imp:
    l1 = "import logger\n"
    flag = True

count = 0
for line in fileinput.FileInput("/home/sa.goyal/Desktop/file.txt", inplace=1):
    if line is None:
        break
    
    if flag == True:
        if len(imp) > 0 and line == imp[-1] + '\n':
            line = line +'\n'+ l1
        elif count == 1 and len(imp) == 0:
            line = l1 + line
        
    t = re.search('print.*',line)
    if t:
        tb = " ".join(t.group().split())
        line = line.replace(t.group(),"logger.info({ss})".format(ss=re.search(' (.*)',tb).group(1)))
    sys.stdout.write(line)
    count += 1

In [1]:
import re
with open("/home/sa.goyal/Desktop/ff.txt",'r') as fh:
    data = fh.read()
    
ll = re.findall("\n.*",data)
for i in ll:
    if i=='\n    cc:':
        cc_index = ll.index(i)
        break
ll.insert(cc_index+1,'\n        - abc')
print ll
exit()
with open("/home/sa.goyal/Desktop/ff.txt",'w') as fh:
    fh.writelines(ll)

['\nrev', '\nerv', '\nemail_report:', '\n    to:', '\n        - rahul@123', '\n        - sumit@123', '\n    cc:', '\n        - abc', '\n        - abc', '\n        - manish@123', '\n        - surbhi@123', '\n    email_error:', '\nrfr', '\nrg', '\n4gv4', '\nrt', '\n']


In [144]:
import re
import sys
import fileinput

with open("/home/sa.goyal/Desktop/ff.txt",'r') as fh:
    data = fh.readlines()
s = '    '
d = '        '
flag = False
f=False
for i in data:
    if i.rstrip()=='    cc: []' or i.rstrip()=='    cc:':
        cc_index = data.index(i)
        break
    elif i.rstrip()=='  cc:' or i.rstrip()=='  cc: []':
        f = True 
        cc_index = data.index(i)
        print cc_index
        break
        
count=0
if f==True:
    s = '  '
    d = '    '

for line in fileinput.FileInput("/home/sa.goyal/Desktop/ff.txt", inplace=1):
    if line is None:
        break

    if line.rstrip() == s + 'cc: []':
        flag = True
        line = s + 'cc:\n' + d + '- abc\n'
    elif count==cc_index+1 and line.rstrip() != d + '- abc' and flag==False:
        line = d + "- abc\n" + line
    
    sys.stdout.write(line)
    count=count+1

7


ernvcok
rev
erv
email_report:
    to:
        - rahul@123
        - sumit@123
    cc:
        - manish@123
        - surbhi@123
    email_error:
rfr
rg
gv4
rt

In [85]:
code

'ernvcok\nrev\nerv\nemail_report:\n    to:\n        - rahul@123\n        - sumit@123\n    cc:\n        - manish@123\n        - surbhi@123\n    email_error:\nrfr\nrg\n4gv4\nrt'

In [86]:
import re
ll = re.findall("\n.*",code)

In [87]:
ll

['\nrev',
 '\nerv',
 '\nemail_report:',
 '\n    to:',
 '\n        - rahul@123',
 '\n        - sumit@123',
 '\n    cc:',
 '\n        - manish@123',
 '\n        - surbhi@123',
 '\n    email_error:',
 '\nrfr',
 '\nrg',
 '\n4gv4',
 '\nrt']

In [89]:
for i in ll:
    if i=='\n    cc:':
        cc_index = ll.index(i)
        break
ll.insert(cc_index+1,'\n        - abc')
ll

['\nrev',
 '\nerv',
 '\nemail_report:',
 '\n    to:',
 '\n        - rahul@123',
 '\n        - sumit@123',
 '\n    cc:',
 '\n        - abc',
 '\n        - abc',
 '\n        - manish@123',
 '\n        - surbhi@123',
 '\n    email_error:',
 '\nrfr',
 '\nrg',
 '\n4gv4',
 '\nrt']

In [32]:
import re
ss = '''v
    r4tv
    <body>cr 
    4t
    vt4
    g
    4g
        </body>
'''
ss

'v\n    r4tv\n    <body>cr \n    4t\n    vt4\n    g\n    4g\n        </body>\n'

In [83]:
s = 'e cvv <fewgwrg 12:34rg  r 56:78:12'

In [87]:
re.match('(.*)<.*(\d\d:\d\d).*(\d\d:\d\d):\d\d',s).group(3)

'56:78'